grant-grant-revoke-user-role documentation (#43)
plantanapp/documentation: docs/Actions/grant-user-role.md

| Action Name                                             | Description                                                        |
| ------------------------------------------------------- | ------------------------------------------------------------------ |
| [Load Users](/docs/Actions/load-user.md)                | Can be used to grant roles to a single user or to a list of users. |
| [User Registration](/docs/Actions/user-registration.md) | Can be used to grant roles to a newly created user.                |

## `Input Parameter Reference`

| Parameter        | Description                                                                                                                                                                                                                                                  | Supports Tokens | Default      | Required |
| ---------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | --------------- | ------------ | -------- |

> | Role             | Select the security role to grant from a list with all the roles in the application. RoleId can also be used as input.                                                                                                                                       | Yes             | Not selected | Yes      |

Reviewer: Also mention that if a user is already in a role you are trying to grant, it will first be removed from that role and then regranted.

> | Other Role Names | Freely input the role name(s) you want to grant. Multiple roles are separated by comma. RoleId can also be used as input.                                                                                                                                    | Yes             | Empty string | No       |

Reviewer: ```suggestion
| Other Role Names | Freely input the role identifiers you want to grant. Multiple roles are separated by a comma or semicolon. The role identifier can be a role name or a role ID.                                                                                                                                    | Yes             | Empty string | No       |
```

> | Role Validity    | Set a validity period to the granted role. The validity period can be done: for a number of days, date interval with or without tokens. <br></br> When the validity period will expire, the user will not have the permissions of that role anymore. | No              | Not selected | No       |

Reviewer: There also is the extend from current validity, offset from now. You should go a bit more into detail about the modes a role validity can be input.

---
id: grant-user-role
title: Grant User Role
sidebar_label: Grant User Role
---

> Audience: [`Business Users`](/docs/audience#business-users)<br/>
> Skill Prerequisites: `Permission Scheme`

## `Typical Use Cases`

- Assign roles to new users on registration
- Give access to resources for specific users
- Temporary give access to users

## `Related Actions`

| Action Name                                             | Description                                                        |
| ------------------------------------------------------- | ------------------------------------------------------------------ |
| [Load Users](/docs/Actions/load-user.md)                | Can be used to grant roles to a single user or to a list of users. |
| [User Registration](/docs/Actions/user-registration.md) | Can be used to grant roles to a newly created user.                |

## `Input Parameter Reference`

> | Parameter        | Description                                                                                    | Supports Variables |

Reviewer: Also, add "Required"( Yes/No) and Default(the default value ) as headers of this table.

Author: Done

| ---------------- | ---------------------------------------------------------------------------------------------- | ------------------ |

> | Role             | Select the security role, from existing roles, to grant/revoke.                                | Yes                |

Reviewer 2: ```suggestion
| Role             | Select the security role to grant from a list with all the roles in the application.                                | Yes                |
```

> | Other Role Names | Freely input the role name(s) you want to grant/revoke. Multiple roles are separated by comma. | Yes                |

Reviewer 2: ```suggestion
| Other Role Names | Freely input the role name(s) you want to grant. Multiple roles are separated by comma. | Yes                |
```

Reviewer 2: Can I also use Role IDs?

Author: Yes. I mentioned it now

> | Role Validity    | Set a validity period to the granted role.                                                     | No                 |

Reviewer 2: How is the validity expressed? 
What happens when it expires?
How would I extend it?

Reviewer 2: I see some of these questions are in the FAQ. Let's move the response here.

Author: Updated with the answers through text

## `Examples`

### `1. Grant a User Role`

The action below register a new user and grants it the '[Citizen Developers]' Role. [Import it](/docs/Actions/Import-actions) into your application to see it in action.

```json
{
    "Title": "Execute Actions",
    "ActionType": "ExecuteActions",

>     "Description": "grant Citizen Developers Role to testuser",

Reviewer 2: ```suggestion
    "Description": "Grant the Citizen Developers role to testuser",
```

                        "IsExpression": false,
                        "Parameters": {}
                    },
                    "ExtensionDays": "",
                    "StartDate": {
                        "Date": ""
                    },
                    "StartDateToken": "",
                    "ExpireDate": {
                        "Date": ""
                    },
                    "ExpireDateToken": "",
                    "RoleExpiration": ""
                },
                "ActionType": "GrantUserRole",
                "$_isOpen": false,
                "$_isLoaded": true,
                "$_isFocus": true,
                "Definition": {
                    "IsClientAction": false,
                    "Settings": {
                        "Group": "User"
                    }
                }
            }
        ]
    }
}
```

### `2. Grant a User Role for a period of time`

The action below register a new user and grants it the '[Citizen Developers]' Role just for 30 days. [Import it](/docs/Actions/Import-actions) into your application to see it in action.

```json
{
    "Title": "Execute Actions",
    "ActionType": "ExecuteActions",

>     "Description": "grant Citizen Developers Role for 30 days to testuser",

Reviewer 2: ```suggestion
    "Description": "Grant the Citizen Developers role for 30 days to testuser",
```

                        "Value": "6",
                        "IsExpression": false,
                        "Parameters": {}
                    },
                    "RoleNames": "",
                    "DateSelectionMode": {
                        "Expression": "",
                        "IsExpression": false,
                        "Parameters": {},
                        "Value": "DateWithOffset"
                    },
                    "ExtensionDays": "",
                    "StartDate": {
                        "Date": ""
                    },
                    "StartDateToken": "",
                    "ExpireDate": {
                        "Date": ""
                    },
                    "ExpireDateToken": "",
                    "RoleExpiration": "30"
                },
                "ActionType": "GrantUserRole",
                "$_isOpen": false,
                "$_isLoaded": true,
                "$_isFocus": true,
                "Definition": {
                    "IsClientAction": false,
                    "Settings": {
                        "Group": "User"
                    }
                }
            }
        ]
    }
}
```

> ## `Frequently asked questions`

Reviewer 2: Remove the FAQ section and move the content in other sections where it falls in naturally.

Author: Done
